### DISCUSSION TITLE
Monitor brightness resets to 100% on Lock Screen or after Sleep

### DISCUSSION THREAD

**Discussion by Malangcow** [MonitorControl/MonitorControl#615] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the bug

In a condition MonitorControl is running and user sets brightness other than 100%. If the Lock Screen function is activated, the monitor brightness resets to 100%. This does not happens when MonitorControl is closed(not running) when Lock Screen function runs.

### Steps to reproduce

1. Run MonitorControl App
2. From the monitor OSD menu(**Set SmartContrast to On**)
3. Set brightness other than 100%(E.g., Set to 50%, The way how the brightness is set doesn't matters(via Monitor control or OSD menu of the monior))
4. Lock Screen(may use keyboard ^⌘Q)
5. The Brightness of the monitor become 100%
(Note. When I check `Brightness value` from the monitor OSD menu it says 50%(The value set at step 2.) When I press brightness +1 Key or brightness -1 key, The value changes 51% or 49% and the actual brightness of the screen now matches the brightness value display on the OSD menu.

The resetting issue does not happen either MonitorControl is not running or  `SmartContrast` function of the monitor is  off.

### Expected behavior

After user configures the brightness. The monitor brightness is consistant when `Screen Lock`ed.

### Anything else?

Since the monitor fuction `SmartContrast` is not that smart(slow adaptation/response). User may not want to use that function.  Turning off the function resolves this issue.

### Environment Information (please complete the following information)

```markdown
- macOS version: Big Sur 11.6
- Mac model: MacBook Air(M1, 2020)
- MonitorControl version: Version 3.0.0 Build 3097
- Monitor(s): 1 External monitor(Philips 325M8C, Connection method: Displaylink)
- Apple Silicon/M1 (yes or no): yes
```

**Comment by waydabber** [comment]
Some displays do this. I advise you not to use smart contrast. The new 4.0.0 version has an option to restore last settings via DDC after sleep which might help with displays like this.

Since this info might be useful for others, I'll move this over to discussion.

**Comment by waydabber** [comment]
@Malangcow - are you using a DisplayLink connected display on an M1 mac and still have DDC control? Or did you meant DisplayPort? Thank you!

**Comment by forthrin** [comment]
MacBook Air M1
macOS Sonoma
Apple Cinema Display 30" (A1083)
Software Brightness Control

1. Move Main Bar dropdown slider to any setting under 100% (eg. 50%)
2. Lock screen
3. Unlock screen
4. BUG: Actual software brightness is at 100%, even though slider is correctly at 50%
5. Dragging slider slightly immediately sets software brightness to expected value (50%)

Can the software be fixed so that software brightness is set according to slider automatically after unlocking?

(Is this the same issue as OP?)
